src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -649,6 +649,7 @@ private FunctionStartedMessage CreateStartedMessageWithoutArguments(IFunctionIns
                 FunctionInstanceId = instance.Id,
                 Function = instance.FunctionDescriptor,
                 ParentId = instance.ParentId,
+                MessageId = instance.MessageId,
                 Reason = instance.Reason,
```

Comment:
I was hoping we could piggy-back on this Reason to add some additional context. Is that not possible? I'm not entirely sure how that thing gets populated.

Reply:
I don't think we'll be able to do that since the Reason is bubbled from the trigger as an Enum of type ExecutionReason. MessageId would need to be some kind of Guid or otherwise unique indentifier.

Reply:
Right - that is a higher level concept, like "manually invoked / HostCall", "automatically triggered", etc.